File: Queue/QueueJava/Queue.java
package Queue.QueueJava;


public class Queue {
    private Node front;
    private Node rear;
    Node testFront;
    Node testRear;
    
    public Queue() {
        front = null;
        rear = null;
    }

    public void enqueue(char dato) {
        Node newNode = new Node(dato);
        if( front == null ) {
            front = newNode;
        } else {
            rear.setNode(newNode);
        }
        rear = newNode;
    }
    
    //Prueba para verificar las referencias
    public void test(char dato, char dato2) {
        Node newNode = new Node(dato);
        Node newNode1 = new Node(dato2);
        testFront = newNode;
        testRear = newNode;
        System.out.println(testFront.getNode());
        testRear.setNode(newNode1);
        System.out.println(testFront.getNode().getDato());
    }
    
    public char dequeue() {
        if(front == null && rear == null) {
            System.out.println("La cola esta vacia");
            return ' ';
        } else {
            Node nodoEliminado = front;
            front = front.getNode();
            nodoEliminado.setNode(null);
            return nodoEliminado.getDato();
        }
    }
    
    public void printQueue() {
        if(front == null && rear == null) {
            System.out.println("Cola Vacia");
        } else {
            Node auxiliar = front;
            while(auxiliar != null) {
                System.out.println(auxiliar.getDato());
                auxiliar = auxiliar.getNode();
            }
        }
    }
    
    public char obtener() {
        if(front == null) {
            return ' ';
        }
        
        return front.getDato();
    }
}

File: Queue/QueuePython/Queue.py
from Node import Node
class Queue:

    __front = None
    __rear = None

    def __init__(self):
        pass


    def enqueue(self, dato):
        newNode = Node(dato)
        if (self.__front == None):
            self.__front = newNode
        else:
            self.__rear.setNode(newNode)
        self.__rear = newNode
        
    def dequeue(self):
        temp = self.__front
        self.__front = self.__front.getNode()
        return temp.getDato()
    
    def print_queue(self):
        temp = self.__front
        while(temp != None):
            print(temp.getDato())
            temp = temp.getNode()
